packages/next-auth/src/providers/hubspot.ts

```diff
@@ -0,0 +1,105 @@
+import type { OAuthConfig, OAuthUserConfig } from "."
+
+interface HubSpotProfile extends Record<string, any> {
+
+  // TODO: figure out additional fields, for now using 
+  // https://legacydocs.hubspot.com/docs/methods/oauth2/get-access-token-information
+
+  user: string,
+  user_id: string,
+
+  hub_domain: string,
+  hub_id: string,
+}
+
+
+const HubSpotConfig = {
+  authorizationUrl: "https://app.hubspot.com/oauth/authorize",
+  tokenUrl: "https://api.hubapi.com/oauth/v1/token",
+  profileUrl: "https://api.hubapi.com/oauth/v1/access-tokens"
+}
+
+export default function HubSpot<P extends HubSpotProfile>(
+  options: OAuthUserConfig<P> & { "redirectURL": string }
```

Comment:
You don't need `redirectURL` here:
- In `authorization`: no need `redirect_uri`, I've just tested without it and it still works
- In `token`: you can use `context.provider.callbackUrl`

Reply:
@ThangHuuVu this is more to make the developer aware that it's a required field as per HubSpot. The request won't work without it. Callback is more of a client set redirect from my understanding of these terms. I also wanted it part of the configuration and environment variables  for the same reason.

Reply:
@Miciurash sorry for the delayed response 👀
Following is what I suggested, just want to make sure we're on the same page:
```ts
    authorization: {
      url: HubSpotConfig.authorizationUrl,
      params: {
        scope: "oauth",
        // redirect_uri: options.redirectURL, You don't need to pass redirect_uri here, 
// NextAuth.js automatically appends it for you in the authorization request.
        client_id: options.clientId,
      },

    },
    token: {
      url: HubSpotConfig.tokenUrl,
      async request(context) {

        if (!context.params.code) {
          throw new Error("Missing HubSpot authentication code in the callback request parameters")
        }
        const urlParams = new URLSearchParams({
          client_id: options.clientId,
          client_secret: options.clientSecret,
          grant_type: "authorization_code",
          redirect_uri: context.provider.callbackUrl, // instead of options.redirectURL
          code: context.params.code,
        });
        
```

The value of the callback URL will be `http://localhost:3000/api/auth/callback/hubspot`
I think it's best to limit the number of configuration and instead reuse what's already there.

Reply:
Got it @ThangHuuVu, thank you for the example! I fixed it.